include/boost/json/detail/basic_parser.hpp

```diff
@@ -108,59 +100,110 @@ class basic_parser
     bool more_; // false for final buffer
     bool complete_ = false; // true on complete parse
     bool is_key_;
+    const char* end_;
     parse_options opt_;
 
     inline static bool is_control(char c) noexcept;
     inline static char hex_digit(char c) noexcept;
+    
     inline void reserve();
-    inline void suspend(state st);
-    inline void suspend(state st, number const& num);
-    inline bool skip_white(const_stream& cs);
+
+    BOOST_NOINLINE
+    inline
+    std::nullptr_t
+    propagate(state st);
+
+    BOOST_NOINLINE
+    inline
+    std::nullptr_t
+    fail(const char* p) noexcept;
+
+    BOOST_NOINLINE
+    inline
+    std::nullptr_t
+    fail(
+        const char* p, 
+        error err) noexcept;
+
+    BOOST_NOINLINE
+    inline
+    std::nullptr_t
+    partial_if_more(
+        const char* p, 
+        state st);
+
+    BOOST_NOINLINE
+    inline
```

Comment:
why `BOOST_NOINLINE` followed by `inline` ?

Reply:
We want the ODR benefits of the `inline` keyword (multiple definitions), but we don't actually want the function to be inlined.